[cxo05] iP

cxo05 opened the pull request:
# Horo
> “Your mind is for having ideas, not holding them.” – David Allen

## GitHub flavored markdown (GFM)

|    Feature | Support              |
| ---------: | :------------------- |
| ~~strikethrough~~ 		 | 100%                 |
|        *italics*| 100%                 |
| **bold**|100%|
| [link](https://example.com) |100%|

- [ ] task list
- [x] checked item

- Title
	  - SubTitle
	  - SubTitle
	  - SubTitle

### Code

```js
import React from "react"
import ReactDOM from "react-dom"
import ReactMarkdown from "react-markdown"
import rehypeHighlight from "rehype-highlight"

ReactDOM.render(
  <ReactMarkdown rehypePlugins={[rehypeHighlight]}>
    {"# Your markdown here"}
  </ReactMarkdown>,
  document.querySelector("#content"),
)
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

cxo05 committed:
Rename ChatBot

cxo05 committed:
Implement skeletal version

cxo05 committed:
Implement echo

cxo05 committed:
Implement text storing

cxo05 committed:
Add mark as done

cxo05 committed:
Add Todo, Events and Deadlines

cxo05 committed:
Add testing

cxo05 committed:
Handle errors, update tests

cxo05 committed:
Implement delete, update tests

cxo05 committed:
Use enums for command parsing

cxo05 committed:
Add task file loading and parsing

cxo05 committed:
Update data file when handling tasks

cxo05 committed:
Merge branch 'branch-Level-7'

cxo05 committed:
Replace string with LocalDateTime

cxo05 committed:
Merge branch 'branch-Level-8'

cxo05 committed:
Refactor code

cxo05 committed:
Update tests

cxo05 committed:
Split into packages

cxo05 committed:
Update test file path

cxo05 committed:
Merge branch 'add-gradle-support'

cxo05 committed:
Update build.gradle

cxo05 committed:
Implement JUnit tests

cxo05 committed:
Fix task adding not being saved

cxo05 committed:
Add some javadocs

cxo05 committed:
Update coding standard

cxo05 committed:
Add find command

cxo05 committed:
Merge branch 'branch-A-JavaDoc'

cxo05 committed:
Merge branch 'branch-A-CodingStandard'

cxo05 committed:
Merge branch 'branch-Level-9'

cxo05 committed:
Remove text UI tests

cxo05 committed:
Add javafx to gradle

cxo05 committed:
Implement javafx dialogbox

cxo05 committed:
Refactor command parsing

cxo05 committed:
Use fxml and implement their controllers

cxo05 committed:
Merge branch 'branch-Level-10'

cxo05 committed:
Add checkstyle

cxo05 committed:
Modify checkstyle

cxo05 committed:
Merge branch 'branch-A-CheckStyle'

cxo05 committed:
Suppress javadoc warnings

yucongkoo commented on src/main/java/horo/Parser.java:
```diff
@@ -0,0 +1,103 @@
+      case "E":
+        t = new Event(arguments[2], arguments[3], arguments[4]);
+        break;
+      default:
```
I think the case statements should be at the same indentation level as switch statement.

cxo05 replied:
Noted

yucongkoo commented on src/main/java/horo/Parser.java:
```diff
@@ -0,0 +1,103 @@
+        c = new AddEventCommand(input);
+        break;
+      case "find":
+        c = new FindCommand(input);
```
I think the case statements should be at the same indentation level as switch statement.

cxo05 replied:
Noted

yucongkoo commented on src/main/java/horo/Ui.java:
```diff
@@ -0,0 +1,54 @@
+        + " event <description> /from <time> /to <time>\n"
+        + " list\n"
+        + " mark <number>\n"
+        + " unmark <number>\n"
```
Would indenting the following + expressions with 8 spaces make the code follow the coding standard more?

cxo05 replied:
Good point. Will be using text blocks instead as with 8 spaces + long strings, its hard to read.

cxo05 replied:
Nvrmind its not supported in java 11

yucongkoo reviewed (commented):
Overall the code was very neat and easy to understand. Good job 👍

RSXIX commented on src/main/java/horo/data/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+package horo.data;
+
+import java.time.LocalDateTime;
```
Good use of avoiding wildcard imports

RSXIX commented on src/main/java/horo/data/Event.java:
```diff
@@ -0,0 +1,50 @@
+
+  @Override
+  public String toString() {
+    return "[E]"
```
Good consistent use of string formatting, easy to read

RSXIX commented on src/main/java/horo/data/Event.java:
```diff
@@ -0,0 +1,50 @@
+  @Override
+  public String getDataString() {
+    return "D,"
+        + (super.isDone() ? "1" : "0") + ","
```
Better to leave a space for + " , "

cxo05 replied:
Not sure what you mean

RSXIX commented on src/main/java/horo/data/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+    return matchedTasks;
+  }
+
+  public ArrayList<Task> getTasks() {
```
I do feel this does violate OOP, so maybe avoid a getter. I tried to not allow anyone to directly access the task in my list so I had to make a few more methods for that.

cxo05 replied:
Not really. The tasks themselves have getters and setters. The getTasks function only allows reading, you cant modify actual taskList.

RSXIX commented on src/main/java/horo/Parser.java:
```diff
@@ -0,0 +1,103 @@
+ */
+public abstract class Parser {
+
+  private static final Pattern commandPattern = Pattern
```
Might be useful to add a comment here to explain what this does

RSXIX reviewed (commented):
Overall I found the code to be quite professional. I love how you group the classes together in packages, but maybe you could reorganize the order of the packages so it makes more logical sense. Consistent use of coding conventions was definitely upheld and I can tell you have experience.

ylyma commented on src/main/java/horo/Ui.java:
```diff
@@ -0,0 +1,54 @@
+
+    horoOutput(getWelcome());
+  }
+
```
Can try adding some Javadocs to the public methods here?

ylyma commented on src/main/java/horo/data/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+  }
+
+  public void showTasks() {
+
```
What does this method do?

cxo05 replied:
Removed

ylyma reviewed (commented):
Good job! Just some minor changes in adherence to the coding standard.

cxo05 committed:
Add javadocs

Javadocs for
 - commandPattern
 - Ui user and horo output functions

Removed unused function

cxo05 committed:
Add Assertion

I dont really see the point of using assertions at this point in time where there is no logic to test and most "logic" testing already has error handling.

cxo05 committed:
Add javadocs

cxo05 committed:
Update fxml

cxo05 committed:
Merge pull request #1 from cxo05/branch-A-Assertions

Add Assertion

cxo05 committed:
Merge pull request #2 from cxo05/branch-A-CodeQuality

Improve code quality

cxo05 committed:
Major Refactor

Moved commands into their separate folders
Added Expense data type
Added AddExpenseCommand
Added ListExpenseCommand

cxo05 committed:
Add delete expense command

cxo05 committed:
Update command display list

cxo05 committed:
Merge branch 'branch-D-Expenses'

cxo05 committed:
Update GUI

cxo05 committed:
Fix delete bug

cxo05 committed:
Add user guide

cxo05 committed:
Update docs

cxo05 committed:
Update gradle and files for fat jar
